File: app/src/main/java/com/example/newspal/fragments/SavedFullNewsFragment.java
package com.example.newspal.fragments;

import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;

import com.bumptech.glide.Glide;
import com.example.newspal.R;
import com.example.newspal.model.Article;
import com.example.newspal.viewmodel.MainActivityViewModel;

public class SavedFullNewsFragment extends Fragment {
    private Article selectedNews;
    private ImageView fullNewsImage;
    private TextView fullNewsTitle,fullNewsDescription;
    private MainActivityViewModel mainActivityViewModel;
    private Button fullNewsBtn;
    private ImageButton delBtn,shareBtn;
    public SavedFullNewsFragment() {
        // Required empty public constructor

    }


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mainActivityViewModel = new ViewModelProvider(this).get(MainActivityViewModel.class);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_saved_full_news, container, false);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        Article selectedNews = SavedFullNewsFragmentArgs.fromBundle(getArguments()).getSelectedNews();


        delBtn = view.findViewById(R.id.delBtn);

        shareBtn = view.findViewById(R.id.shareBtn);
        int nightModeFlags = getContext().getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;
        if(nightModeFlags== Configuration.UI_MODE_NIGHT_YES)
        {
            delBtn.setImageResource(R.drawable.ic_baseline_delete_forever_light_24);
            shareBtn.setImageResource(R.drawable.ic_baseline_share_light_24);
        }

        delBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(getActivity(), "Deleted!!", Toast.LENGTH_SHORT).show();
                mainActivityViewModel.deleteArticle(selectedNews);

            }
        });
        fullNewsImage = view.findViewById(R.id.fullNewsImage);
        fullNewsDescription = view.findViewById(R.id.fullNewsDescription);
        fullNewsTitle = view.findViewById(R.id.fullNewsTitle);
        fullNewsBtn = view.findViewById(R.id.fullNewsBtn);
        if(selectedNews.getUrlToImage()==null)
            Glide.with(view)
                    .load(R.drawable.image)
                    .into(fullNewsImage);
        else
            Glide.with(view)
                    .load(selectedNews.getUrlToImage())
                    .into(fullNewsImage);
        fullNewsTitle.setText(selectedNews.getTitle());
        fullNewsDescription.setText(selectedNews.getDescription());
        fullNewsBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                NavController navController = NavHostFragment.findNavController(getParentFragment());
                navController.navigate(SavedFullNewsFragmentDirections.actionSavedFullNewsFragmentToWebViewFragment(selectedNews.getUrl()));
            }
        });
        shareBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent i = new Intent(Intent.ACTION_SEND);
                i.setType("text/plain");
                i.putExtra(Intent.EXTRA_SUBJECT, "Sharing URL");
                i.putExtra(Intent.EXTRA_TEXT, selectedNews.getUrl());
                startActivity(Intent.createChooser(i, "Share URL"));
            }
        });
    }
}

File: app/src/main/java/com/example/newspal/fragments/SavedFullNewsFragment.kt
package com.example.newspal.fragments

import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.core.net.toUri
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.navigation.fragment.findNavController
import coil.load
import com.example.newspal.NewsApplication
import com.example.newspal.R
import com.example.newspal.databinding.FragmentSavedFullNewsBinding
import com.example.newspal.viewmodels.MainViewModel
import com.example.newspal.viewmodels.MainViewModelFactory

class SavedFullNewsFragment : Fragment() {

    val viewModel: MainViewModel by viewModels {
        MainViewModelFactory(
            (activity?.application as NewsApplication).database,
        )
    }

    private var _binding: FragmentSavedFullNewsBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment
        _binding = FragmentSavedFullNewsBinding.inflate(inflater,container,false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val nightModeFlags = requireContext().resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK
        if (nightModeFlags == Configuration.UI_MODE_NIGHT_YES) {
            binding.delBtn.setImageResource(R.drawable.ic_baseline_delete_forever_light_24)
            binding.shareBtn.setImageResource(R.drawable.ic_baseline_share_light_24)
        }
        val article = SavedFullNewsFragmentArgs.fromBundle(requireArguments()).selectedNews
        binding.delBtn.setOnClickListener {
            Toast.makeText(activity, "Deleted!!", Toast.LENGTH_SHORT).show()
           viewModel.deleteArticle(article)
            findNavController().navigateUp()
        }
        binding.fullNewsTitle.text = article.title.toString()
        binding.fullNewsDescription.text = article.description.toString()
        val imgUrl = article.urlToImage
        if(imgUrl==null) {
            binding.fullNewsImage.load(R.drawable.ic_broken_image)
        } else {
            val imgUri = imgUrl.toUri().buildUpon().scheme("https").build()
            binding.fullNewsImage.load(imgUri) {
                placeholder(R.drawable.loading_animation)
            }
        }
        binding.fullNewsBtn.setOnClickListener {
            this.findNavController().navigate(SavedFullNewsFragmentDirections.actionSavedFullNewsFragmentToWebViewFragment(article.url!!))
        }
        binding.shareBtn.setOnClickListener {
            val intent = Intent(Intent.ACTION_SEND)
            intent.type = "text/plain"
            intent.putExtra(Intent.EXTRA_SUBJECT, "Sharing URL")
            intent.putExtra(Intent.EXTRA_TEXT, article.url)
            startActivity(Intent.createChooser(intent, "Share URL"))
        }
    }
}